### PR TITLE
feat: update peer deps for react

### PR DESCRIPTION
**Pull request by ivanoats**
Updating the peer dependencies for React

Resolves https://github.com/google-map-react/google-map-react/issues/1012

### CODE REVIEW

**Review by mrantivirus — APPROVED** [review]

**Comment by mrantivirus** [comment]
@karldanninger @itsmichaeldiego - Can we get an update on this.

**Review by ovandenbosch — APPROVED** [review]
Please can we get a merge on this.

**Review by momasz — APPROVED** [review]
would be awesome to get this merged 🙌

**Comment by mrantivirus** [comment]
I got an update from the maintainer. He is away from the computer, but he'll take a view if this pull request, but he wont be able to deploy until early June.

**Comment by JulesPatry** [comment]
Really hoping this gets merged in soon!

File: package.json
Comment (jdwillemse) [inline package.json:37]: To allow as many users as possible to use this 
```
"react": "^16.0.0 || ^17.0.0",
"react-dom": "^16.0.0 || ^17.0.0",
```

File: package.json
Comment (jdwillemse) [inline package.json:59]: To allow as many users as possible to use this 
```
"react": "^16.0.0 || ^17.0.0",
"react-dom": "^16.0.0 || ^17.0.0",
```

**Review by jdwillemse — APPROVED** [review]

**Comment by iyobo** [comment]
Let's get this in please

File: package.json
Comment (ivanoats) [reply package.json:37]: will do

**Comment by loopdream** [comment]
Please can we get this in 🙏🏻

**Comment by iyobo** [comment]
For anyone blocked by this, it is best to use the branch with the fix
directly package.json.

This should unblock you until the actual merge.

**Comment by itsmichaeldiego** [comment]
Hey folks! I am back. Just published it in 2.1.10, let me know!

**Pull request merged**
